### DISCUSSION TITLE
How to understand these labels supported in Pulseq?

### DISCUSSION THREAD

**Discussion by zhengliuX** [pulseq/pulseq#57] [Q&A] [answered]
Dear all,
    I am trying to understand the labels supported in Pulseq, most of them are clear to understand for me, but still, there are some that I'm not sure about. I will list all the labels and my understanding below, correct me if I'm wrong.(For the labels I am not sure, I add ??)

```
SLC: Slice for 2D, slice phase encoding for 3D
SEG: Segment, Split one scan to multiple segments
REP: Repetition, the repetition number of current acq
AVG: Average
SET: Set, flexible meaning??
ECO: Echo, Used for Multi-echo sequence
PHS: Phase Correction??
LIN: PE index
PAR: a flag for Parallel imaging??
NAV: a flag for navigator
REV: a flag for reversion, common in EPI
SMS: a flag for Simultaneous Multi-Slice
REF: Reference line
IMA: Image Line
NOISE: Noise line
PMC: a flag for Motion correction (MoCo or PMC)
```

**Comment by m-a-x-i-m-z** [comment, marked as answer]
Good point, we should add this to the DOCS section of the main website. 

In fact we do provide some useful comments inside the function mr.gerSupportedLabels(): 

```
function supported_labels = getSupportedLabels()
% auxilary function

supported_labels={'SLC','SEG','REP','AVG','SET','ECO','PHS','LIN','PAR','NAV','REV','SMS', ... # these label values are copied to the corresponding MDH field on Siemens
    'REF', 'IMA', ... # For parallel imaging
    'NOISE', ... % noise adjust scan, for iPAT acceleration
        'PMC', ... # for MoCo/PMC Pulseq version to recognize blocks that can be prospectively corrected for motion 
        'NOROT','NOPOS','NOSCL', ... # instruct the interpreter to ignore the position, rotation or scaling of the FOV specified on the UI 
        'ONCE', ... # a 3-state flag that instructs the interpreter to alter the sequence when executing multiple repeats as follows: blocks with ONCE==0 are executed on every repetition; ONCE==1: only the first repetition; ONCE==2: only the last repetition
        'TRID' ... # an integer ID of the TR (sequence segment) used by the GE interpreter to optimize the execution on the scanner
    };

end
````

However I agree that more information might be useful. Here is the first attempt:
| Label | Type | Data Mapping | Meaning  |
|--------|-------|--------------------|----------|
| SLC | counter | yes | slice counter (or slab counter for 3D multi-slab sequences) |
|SEG | counter | yes | segment counter e.g. for segmented FLASH or EPI |
| REP | counter | yes | repetition counter |
| AVG | counter | yes | averaging counter | 
| SET | counter | yes | flexible counter without firm assignment | 
| ECO | counter | yes | echo counter in multi-echo sequences | 
| PHS | counter | yes | cardiac phase counter |
| LIN | counter | yes | line counter in 2D and 3D acquisitions | 
| PAR | counter | yes | partition counter; it counts phase encoding steps in the 2nd (through-slab) phase encoding direction in 3D sequences |
| ACQ | counter | yes | spectroscopic acquisition counter | 
| NAV | flag | yes | navigator data flag | 
| REV | flag | yes | flag indicating that the readout direction is reversed | 
| SMS | flag | yes | simultaneous multi-slice (SMS) acquisition |
| REF | flag | yes | parallel imaging flag indicating reference / auto-calibration data |
| IMA | flag | yes | parallel imaging flag indicating imaging data within the ACS region |
| NOISE | flag | yes | flag for the noise adjust scan e.g for the parallel imaging acceleration |
| PMC | flag | no | flag for the MoCo/PMC Pulseq version marking blocks that can/should be prospectively corrected for motion | 
| NOROT | flag | no | instructs the interpreter to ignore the rotation of the FOV specified on the UI for the given block(s) |
| NOPOS | flag | no | instructs the interpreter to ignore the the FOV offset specified on the UI for the given block(s) |
| NOSCL | flag | no | instructs the interpreter to ignore the scaling of the FOV specified on the UI for the given block(s) |
| ONCE | 3-state flag| no | a 3-state flag that instructs the interpreter to alter the sequence when executing multiple repeats as follows: blocks with ONCE==0 are executed on every repetition; ONCE==1: only on the first repetition; ONCE==2: only on the last repetition |
| TRID | counter | no | marks the beginning of a repeatable module in the sequence (e.g. TR); modules with different timing should be assigned different TRIDs |

**Reply by zhengliuX** [reply]
Professor, does PAR(partition counter) means slice encoding counter? Phase encoding is the 1st encoding, and slice encoding is the 2nd.

And what does Data Mapping in the table mean?

**Reply by m-a-x-i-m-z** [reply]
Hi Zheng Liu, 

the PAR counter only refers to the 2nd phase encoding direction, along which the FFT is applied. SLC counter is used for separately excited slices (or slabs in case of 3D imaging). In a single-slab 3D sequence SLC counter always stays at 0 and the PAR counter is used to mark all the encoding steps in the slice direction. 

The column "data mapping" in the table above refers to the fact, that not all labels supported by Pulseq are related to the labeling of raw data. For example TRID or PMC do not get copied to raw data headers or flags in any way.

**Reply by zhengliuX** [reply]
Thanks a lot, professor. 

Also, does this mean we can design a multi-slabs sequence with Pulseq?

**Reply by m-a-x-i-m-z** [reply]
of course we can design multi-slab sequences with Pulseq! ;-)

**Reply by zhengliuX** [reply]
Dear all,

Hi, is there any multi-slab pulseq demo available now?

Best,
Zheng Liu

**Reply by QingpingChen1201** [reply]
> Dear all,
> 
> Hi, is there any multi-slab pulseq demo available now?
> 
> Best, Zheng Liu

Hi Zheng,
as far as I know, we don't have example sequences for multi-slab excitation yet. All of the example sequences are here: https://github.com/pulseq/pulseq/tree/master/matlab/demoSeq
Best regards,
QC

**Reply by zhengliuX** [reply]
Thanks！😄

**Comment by QingpingChen1201** [comment]
Dear all,
I would like to add a bit for the "IMA" and "REF" labels. The "IMA" label is used to mark the imaging lines within the auto-calibration signal (ACS) region. The imaging lines outside the ACS region are not labeled with "IMA" and "REF". Hope the figure below can make it more clear.
Best regards,
Qingping
![GRAPPA_label](https://github.com/user-attachments/assets/6ac25c1a-0a73-47d8-a31c-cecd6a6c77fe)

**Reply by zhengliuX** [reply]
So, there is no label for Imaging lines outside ACS, or should I see all lines without labels as Imaging lines by default?

**Reply by QingpingChen1201** [reply]
There are no REF & IMA labels for the imaging lines outside ACS - you can set the REF and IMA flags to 0. By default, the REF and IMA labels are set to 0. For more information, please refer to our example sequence: [matlab/demoSeq/writeGradientEcho_grappa.m](https://github.com/pulseq/pulseq/blob/master/matlab/demoSeq/writeGradientEcho_grappa.m)
Best regards,
Qingping

**Reply by zhengliuX** [reply]
Thanks a lot!

**Comment by mcencini** [comment]
Hi, I wanted to ask what is the intended use of the new 'OFF' label. Can I use it to mark the steady state prep scans and prospective motion correction navigator scans? Would that be consistent with their definition?

**Comment by m-a-x-i-m-z** [comment]
Hi!

OFF label is a new one which on Siemens allows to make the acquired (!) ADC data invisible for the ONLINE integrated image reconstruction. However, the data are still stored in the raw data file and are available for an off-line reconstruction. 

Hope this helps!

--M.
